docs/workflow/testing/mono/testing.md

```diff
@@ -1,54 +1,83 @@
+* Build and run runtime tests for Webassembly. From the root directory run the following command:
+```
+src/tests/build.sh -skipstressdependencies -excludemonofailures os Browser wasm <Release/Debug>
+```
+    * From the last few lines of the build log, you will see something like this
+```
+--------------------------------------------------
+ Example run.sh command
 
-1. Build and set the RuntimeFlavor
+ src/tests/run.sh --coreOverlayDir=<repo_root>artifacts/tests/coreclr/Browser.wasm.Release/Tests/Core_Root --testNativeBinDir=<repo_root>/artifacts/obj/coreclr/Browser.wasm.Release/tests --testRootDir=<repo_root>/artifacts/tests/coreclr/Browser.wasm.Release --copyNativeTestBin Release
+--------------------------------------------------
+```
+    * To run the tests, copy that command and add `wasm` at the end.
 
-```bash
-./build.sh /p:RuntimeFlavor=mono
+* Build and run Runtime Tests for Android x64. From the root directory run the following command:
+```
+src/tests/build.sh -skipstressdependencies -excludemonofailures os Android x64 <Release/Debug>
 ```
-or on Windows
-```bat
-build.cmd /p:RuntimeFlavor=mono
+    * From the last few lines of the build log, you will see something like this
 ```
+--------------------------------------------------
+ Example run.sh command
 
-2. cd into the test library of your choice (`cd src/libraries/<library>/tests`)
+ src/tests/run.sh --coreOverlayDir=<repo_root>artifacts/tests/coreclr/Browser.wasm.Release/Tests/Core_Root --testNativeBinDir=<repo_root>/artifacts/obj/coreclr/Browser.wasm.Release/tests --testRootDir=<repo_root>/artifacts/tests/coreclr/Browser.wasm.Release --copyNativeTestBin Release
+--------------------------------------------------
+```
+    * To run the tests, copy that command and add `Android` at the end.
+
+* For more details about the nature of the runtime tests, please refer to [Coreclr testing documents](../coreclr)
+
+## Running Library Tests
+* Build and run library tests against Mono JIT:
 
-3. Run the tests
+1. cd into the test library of your choice (`cd src/libraries/<library>/tests`)
+
+2. Run the tests
 
 ```
-dotnet build /t:Test /p:RuntimeFlavor=mono
+dotnet build /t:Test /p:RuntimeFlavor=mono /p:Configuration=<Release/Debug>
 ```
 
-# Patching Local dotnet (.dotnet-mono)
-Another way to test mono out is by 'patching' a local dotnet with our runtime bits.  This is a good way to write simple
-test programs and get a glimpse of how mono will work with the dotnet tooling.
+* Build and run library tests against Webassembly. See instructions for [Testing Webassembly](../libraries/testing-wasm.md)
+
+* Build and run library tests against Android. See instructions for [Testing Android](../libraries/testing-android.md)
 
-To generate a local .dotnet-mono, execute this command:
+* Build and run library tests against iOS. See instructions for [Testing iOS](../libraries/testing-apple.md)
 
+# Test with sample program
+There is a HelloWorld sample program lives at
```

Comment:
I would mention that there are a variety of samples. Is there anything special to do for the other ones, or is `make run` sufficient in every case?

Reply:
It is upto how the Makefile was created. There are ones needs `make build && make run`. I will add instructions to them here.